packages/material-ui/src/styles/ThemeProvider.test.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { expect } from 'chai';
+import { createClientRender } from 'test/utils';
+import { useTheme } from '@material-ui/styles';
+import { ThemeContext } from '@material-ui/styled-engine';
+import ThemeProvider from './ThemeProvider';
+
+describe('ThemeProvider', () => {
+  const render = createClientRender();
+
+  it('should provide the theme to the mui theme context', () => {
+    const ref = React.createRef();
+    const text = () => ref.current.textContent;
+    function Test() {
+      const theme = useTheme();
+
+      return <span ref={ref}>{theme.foo}</span>;
+    }
+
+    render(
+      <ThemeProvider theme={{ foo: 'foo' }}>
+        <Test />
+      </ThemeProvider>,
+    );
+    expect(text()).to.equal('foo');
```

Comment:
nit: This is an existing pattern in our tests that probably pre-dates testing-library. It's generally over-engineered. We can use a lot less indirection and assert on the actual shape with e.g.

```jsx
    let theme;
    function Test() {
      theme = useTheme();

      return null;
    }

    render(
      <ThemeProvider theme={{ foo: 'foo' }}>
        <Test />
      </ThemeProvider>,
    );
    expect(theme).to.deep.equal(THE_SHAPE_OF_THE_THEME);

Reply:
Done 👍